src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,46 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+
+  @each $day, $margin in $days {
+    &-start-day--#{$day} &__day:first-child {
+      margin-left: $margin;
+    }
+  }
+
+  &__day {
+    width: $squareWidth;
+    height: $squareWidth;
+    background-color: #eee;
+    border: 1px solid black;
+    position: relative;
+    cursor: pointer;
+    transition: 0.5s;
+
+    @for $index from 1 through 31 {
+      &:nth-child(#{$index}):before {
+
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        position: absolute;
+        width: 100%;
+        height: 100%;
```

Comment:
add all these styles outside the loop, leave in the loop styles that depend only on the iteration of the loop